src/controllers/trustpub/tokens/exchange/mod.rs

```diff
@@ -0,0 +1,146 @@
+use super::json;
+use crate::app::AppState;
+use crate::util::errors::{AppResult, bad_request, server_error};
+use axum::Json;
+use crates_io_database::models::trustpub::{NewToken, NewUsedJti};
+use crates_io_database::schema::trustpub_configs_github;
+use crates_io_diesel_helpers::lower;
+use crates_io_trustpub::access_token::AccessToken;
+use crates_io_trustpub::github::{GITHUB_ISSUER_URL, GitHubClaims};
+use crates_io_trustpub::unverified::UnverifiedClaims;
+use diesel::prelude::*;
+use diesel::result::DatabaseErrorKind::UniqueViolation;
+use diesel::result::Error::DatabaseError;
+use diesel_async::scoped_futures::ScopedFutureExt;
+use diesel_async::{AsyncConnection, RunQueryDsl};
+use secrecy::ExposeSecret;
+
+#[cfg(test)]
+mod tests;
+
+/// Exchange an OIDC token for a temporary access token.
+#[utoipa::path(
+    put,
```

Comment:
To be compliant with [RFC 8693](https://datatracker.ietf.org/doc/html/rfc8693#name-request), this should be done via a POST.

Reply:
seems like that RFC requires quite a few other things like using `application/x-www-form-urlencoded` (which PyPI also doesn't do). I wasn't aware that we were supposed to follow that RFC? 🤔 

the rest of the API generally uses `PUT` to create resources and `POST` for editing existing resources, which is why I chose `PUT` here.

Reply:
Yeah, I don't believe PyPI follows that RFC either (at least, I didn't follow it while implementing Trusted Publishing 😅).

IMO the state/parameters used here are pretty distinct from the standard OAuth2 Token Exchange anyways, so deviating here seems like not a huge deal.

Reply:
Ah, fair enough. :) My opinion here is loosely held, there's not hard requirement to implement this to the RFC. I don't believe I mentioned this RFC in the Rust RFC either. 😅

Reply:
I think matching PyPI is probably more sensible right now than RFC 8693. It doesn't seem like implementing that RFC in the future would prevent us from implementing this the way we have right now anyway, so this is a changeable decision.